### PR TITLE
KAFKA-3400: Fix describe topic in case there are zero partitions.

### PR DESCRIPTION
**Pull request by SinghAsDev**

### CODE REVIEW

**Comment by SinghAsDev** [comment]
@gwenshap mind taking a look?

**Comment by gwenshap** [comment]
Sorry, but when would a topic have 0 partitions?

**Comment by gwenshap** [comment]
Did you reproduce the original issue? because the code you fixed doesn't seem to be the same one where the bug occurred...

If you did, can you post sample output?

**Comment by SinghAsDev** [comment]
@gwenshap I have not reproduced the issue, but it is evident from the stacktrace on the JIRA. NoSuchElementException gets thrown in `topicPartitionAssignment.head._2.size`. This happens when topic exists, but there are no partitions. I think you are looking at wrong branch, the ST is for 0.9.0.1 branch, https://github.com/apache/kafka/blob/0.9.0/core/src/main/scala/kafka/admin/TopicCommand.scala#L198. Helps?

File: core/src/main/scala/kafka/admin/TopicCommand.scala [hunk 1]
```diff
@@ -216,7 +219,7 @@ object TopicCommand extends Logging {
               val leader = zkUtils.getLeaderForPartition(topic, partitionId)
               if ((!reportUnderReplicatedPartitions && !reportUnavailablePartitions) ||
                   (reportUnderReplicatedPartitions && inSyncReplicas.size < assignedReplicas.size) ||
-                  (reportUnavailablePartitions && (!leader.isDefined || !liveBrokers.contains(leader.get)))) {
+                  (reportUnavailablePartitions && (leader.isEmpty || !liveBrokers.contains(leader.get)))) {
```
Comment (gwenshap) [inline core/src/main/scala/kafka/admin/TopicCommand.scala:222]: I'm not sure how this is related?

Comment (SinghAsDev) [inline core/src/main/scala/kafka/admin/TopicCommand.scala:222]: This is just to get rid of a compiler warning/suggestion. I can remove this. The actual fix is other part :).

**Comment by SinghAsDev** [comment]
@gwenshap @ijuma mind taking a look at this?

**Comment by ijuma** [comment]
Unclear how a topic can have 0 partitions and we haven't seen any reports in the last 1.5 years. Closing for now.

**Pull request closed without merging**